File: src/test/groovy/com/benjaminsproule/swagger/gradleplugin/test/springmvc/ExtendedTestResourceWithoutClassAnnotation.java
package com.benjaminsproule.swagger.gradleplugin.test.springmvc;

import io.swagger.annotations.ApiOperation;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

public class ExtendedTestResourceWithoutClassAnnotation extends TestResourceWithoutClassAnnotation {

    @ApiOperation("An overriden operation description")
    @RequestMapping(path = "/root/withoutannotation/overriden", method = RequestMethod.GET)
    @Override
    public String overriden() {
        return "";
    }

    @RequestMapping(path = "/root/withoutannotation/overridenWithoutDescription", method = RequestMethod.GET)
    @Override
    public String overridenWithoutDescription() {
        return "";
    }
}

File: scala-tests/src/main/scala/com/benjaminsproule/swagger/gradleplugin/test/scala/springmvc/ExtendedTestResourceWithoutClassAnnotation.scala
package com.benjaminsproule.swagger.gradleplugin.test.scala.springmvc

import io.swagger.annotations.ApiOperation
import org.springframework.web.bind.annotation.{RequestMapping, RequestMethod}

class ExtendedTestResourceWithoutClassAnnotation extends TestResourceWithoutClassAnnotation {

    @ApiOperation("An overriden operation description")
    @RequestMapping(path = Array("/root/withoutannotation/overriden"), method = Array(RequestMethod.GET))
    @Override
    override def overriden(): String = {
        ""
    }

    @RequestMapping(path = Array("/root/withoutannotation/overridenWithoutDescription"), method = Array(RequestMethod.GET))
    @Override
    override def overridenWithoutDescription(): String = {
        ""
    }
}
